### PR TITLE
WebXR world scaling factor

### LINKED ISSUE #12441: World/camera scaling
Introduce world-scale factor to XR to allow travelling through worlds where 1 unit is not 1 meter (but more/less).

Reference topic:  https://forum.babylonjs.com/t/manipulating-player-size-in-webxr/29517

### PR DESCRIPTION
**Pull request by RaananW**
Fixes #12441

The webxr session manager now has a world scaling factor that allows to set the units in which your world is defined. i.e., if, in your world, 1 unit is 1 cm, you would define the factor to be 100. the native XR component will still maintain the 1 unit 1 meter relation, but the data from and to XR are scaled correctly according to the scale factor.

Controllers and hands are scaled as well. So is the teleportation ring.

Note - any other values should be scaled by the dev itself. For example, the ray's distance will be in world scale. Another example - to define gestures for the hands components you would need to use the components' absolute position and scale the distances accordingly.

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14716/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14716/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14716/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14716/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14716/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14716/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14716/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14716/merge#BCU1XR#0

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14716/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

File: packages/dev/core/src/XR/features/WebXRHandTracking.ts [hunk 1]
```diff
@@ -451,6 +456,8 @@ export class WebXRHand implements IDisposable {
             Matrix.FromArrayToRef(this._jointTransformMatrices, jointIdx * 16, this._tempJointMatrix);
             this._tempJointMatrix.decompose(undefined, jointTransform.rotationQuaternion!, jointTransform.position);
 
+            // jointTransform.position.scaleInPlace(xrSessionManager.worldScalingFactor);
```
Comment (carolhmj) [inline packages/dev/core/src/XR/features/WebXRHandTracking.ts:459]: nit: commented code

**Review by carolhmj — APPROVED** [review]

File: packages/dev/core/src/XR/features/WebXRHandTracking.ts (hunk 1 above)
Comment (RaananW) [inline packages/dev/core/src/XR/features/WebXRHandTracking.ts:459]:
```suggestion
```

Comment (RaananW) [reply packages/dev/core/src/XR/features/WebXRHandTracking.ts]: thanks! :-) 
done

**Review by sebavan — APPROVED** [review]

**Comment by bjsplat** [comment]
Visualization tests for webgl2 have failed. If some tests failed because the snapshots do not match, the report can be found at

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14716/merge/testResults/webgl2/index.html

If tests were successful afterwards, this report might not be available anymore.

**Pull request merged**